src/dto/hourly-apportioned-emissions.params.dto.ts

```diff
@@ -8,18 +12,41 @@ import { UnitType } from '../enums/unit-type.enum';
 import { State } from '../enums/state.enum';
 import { IsOrisCode } from '../pipes/is-oris-code.pipe';
 import { StateValidation } from '../pipes/state-validation.pipe';
-import { DateValidation } from '../pipes/date-validation.pipe';
+import { IsIsoFormat } from '../pipes/is-iso-format.pipe';
+import { IsValidDate } from 'src/pipes/is-valid-date.pipe';
+import { IsInDateRange } from '../pipes/is-in-date-range.pipe';
+import { IsDateGreaterThanEqualTo } from '../pipes/is-date-greater.pipe';
 import { IsControlTechnology } from 'src/pipes/is-control-technology.pipe';
 
 export class HourlyApportionedEmissionsParamsDTO extends PaginationDTO {
-  @IsOptional()
-  @ApiPropertyOptional()
-  @Validate(DateValidation)
+  @ApiProperty()
+  @IsInDateRange('1995-01-01', {
+    message: 'Please enter a $property year between 1995 and this year',
+  })
+  @IsValidDate({
+    message: 'Please enter a valid $property in the YYYY-MM-DD format',
+  })
+  @IsIsoFormat({
+    message: 'Please enter the $property in the YYYY-MM-DD format',
+  })
+  @IsDefined()
   beginDate: Date;
 
-  @IsOptional()
-  @ApiPropertyOptional()
-  @Validate(DateValidation)
+  @ApiProperty()
+  @IsDateGreaterThanEqualTo('beginDate', {
+    message:
+      'Please enter an end date that is equal to or greater than the begin date',
+  })
+  @IsInDateRange('1995-01-01', {
```

Comment:
consider making this a true date range check where you can pass a minDate and maxDate. both minDate and maxDate can be optional and if so then the validator only checks to see if >= to minDate or <= maxDate and if both are provided then checks both conditions.